File: auction-app-ui/src/services/bidService.ts
import { createBidRequest } from 'models/request/create/createBidRequest';

import agent from 'lib/agent';
import { Bid } from 'models/bid';

const bidService = {
  getHighestBid: (productId: string) =>
    agent.get<number>(`/product/${productId}/bids/highest`),
  addBid: (createBidRequest: createBidRequest, productId: string) =>
    agent.post(`/product/${productId}/bids`, createBidRequest),
  getUserBids: (userId: string) =>
    agent.get<Bid[]>(`/user/${userId}/bids`),
  getProductBids: (productId: string, params: {}) =>
    agent.get<any>(`product/${productId}/bids`, params),
};

export default bidService;

File: auction-app-backend/src/main/java/com/internship/auctionapp/services/bid/BidService.java
package com.internship.auctionapp.services.bid;

import com.internship.auctionapp.requests.CreateBidRequest;
import com.internship.auctionapp.models.Bid;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.UUID;

public interface BidService {
    Bid addBid(CreateBidRequest createBidRequest);

    List<Bid> getAllBids();

    Double getHighestBidPrice(UUID productId);

    List<Bid> getUserBids(UUID userId);

    Page<Bid> getProductBids(UUID productId, Integer pageNumber);
}
